### PR TITLE
AppHarness: support headless mode and driver selection

### PR DESCRIPTION
**Pull request by masenf**
Using environment variables `APP_HARNESS_HEADLESS` and `APP_HARNESS_DRIVER`, support switching out the browser engine used by default in the AppHarness.

### CODE REVIEW

File: reflex/testing.py [hunk 1]
```diff
@@ -397,7 +397,23 @@ def frontend(self, driver_clz: Optional[Type["WebDriver"]] = None) -> "WebDriver
             )
         if self.frontend_url is None:
             raise RuntimeError("Frontend is not running.")
-        driver = driver_clz() if driver_clz is not None else webdriver.Chrome()
+        want_headless = False
+        options = None
+        if os.environ.get("APP_HARNESS_HEADLESS"):
```
Comment (Lendemor) [inline reflex/testing.py:402]: Do we start adding the `REFLEX_` prefix now for the env vars, or do we use a different PR to change all of them?

File: reflex/testing.py [hunk 2]
```diff
@@ -397,7 +397,23 @@ def frontend(self, driver_clz: Optional[Type["WebDriver"]] = None) -> "WebDriver
...
+            want_headless = True
+        if driver_clz is None:
+            requested_driver = os.environ.get("APP_HARNESS_DRIVER", "Chrome")
```
Comment (Lendemor) [inline reflex/testing.py:405]: same comment as above.

**Review by Lendemor — APPROVED** [review]

File: reflex/testing.py (hunk 1 above)
Comment (picklelo) [reply reflex/testing.py:402]: Let's have a separate PR where we do this for all the env vars
